### PR TITLE
Split partition distribution out of PartitionManager

### LINKED ISSUE #13739: Start raft partitions based on the configuration provided by a configuration manager

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This PR takes resolving partition distribution out of RaftPartitionGroup. Partition distribution is generated by `ClusterTopologyManager` and provided to PartitionManager. Later this ClusterTopologyManager will be able to use the persisted topology if it is available instead of generating one from the static properties.

## Related issues

closes #13739 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by deepthidevaki** [comment]
First two commits are the main changes. Rest of them are updating tests, cleanup unused methods etc.

**Review by lenaschoenburg — APPROVED** [review]
Thanks @deepthidevaki, the refactoring makes sense to me :+1:

File: broker/src/main/java/io/camunda/zeebe/broker/partitioning/topology/ClusterTopology.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.partitioning.topology;
+
+import io.atomix.primitive.partition.PartitionMetadata;
+import java.util.Set;
+
+/**
+ * ClusterTopology describes how partitions are distributed across broker. It doesn't keep track of
+ * the current leader or followers.
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply broker/src/main/java/io/camunda/zeebe/broker/partitioning/topology/ClusterTopology.java:15]: Probably not. I guess we can rename it to PartitionDistribution. :+1:

File: broker/src/test/java/io/camunda/zeebe/broker/partitioning/topology/ClusterTopologyManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            Collectors.toMap(
+                p -> p.id().id(),
+                p ->
+                    p.members().stream()
+                        .map(m -> Integer.valueOf(m.id()))
+                        .collect(Collectors.toSet())));
+  }
+
+  @Test
+  void shouldGenerateFixedDistribution() throws IOException {
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply broker/src/test/java/io/camunda/zeebe/broker/partitioning/topology/ClusterTopologyManagerTest.java:74]: I wanted to add a test to verify that we use configured distribution. I don't think we have any other tests for it.

If the issue is with dependency to yaml, I can remove it and just build the config by hand. I was just lazy to do it :smile:

[the remark this answers is not in the document]
Comment (oleschoenburg) [reply broker/src/test/java/io/camunda/zeebe/broker/partitioning/topology/ClusterTopologyManagerTest.java:74]: Haha no that's okay :+1:

**Comment by deepthidevaki** [comment]
bors merge

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**